Merge_Sort_3_Way/Merge_Sort_3_Way.cpp

```diff
@@ -0,0 +1,176 @@
+
+    while( (j < mid.size()) && (k < right.size()) )
+    {
+        if(mid[j] < right[k])
+        {sorted[l++] = mid[j++];}
+        else
+        {sorted[l++] = right[k++];}
+    }
+
+    //selecting minimum of two vectors when mid vector is empty
+
+    while( (i<left.size()) && (k<right.size()) )
+    {
+      if(left[i] < right[k])
+      {sorted[l++] = left[i++];}
+      else
+      {sorted[l++] = right[k++];}
+    }
+
+    // copying remaining elements from left vector when mid and right vector are empty
+
+    while(i < left.size())
+    {sorted[l++] = left[i++];}
+
+    // copying remaining elements from mid vector when left and right vector are empty
+
+    while(j < mid.size())
+    {sorted[l++] = mid[j++];}
+
+    // copying remaining elements from right vector when mid and left vector are empty
+
+    while(k<right.size())
+    {sorted[l++] = right[k++];}
+}
+
+vector<int> mergeSort3ways(vector<int> &input)
+{
+    if(input.size() == 1)
+    {return input;}
+    else if(input.size() == 2)
+    {
+
+        //two different vectors (element1 and element2) of size 1 to store one element each and no_element vector of size 0
+
+        vector<int>element1,element2,no_element(0);
+        element1.emplace_back(input.at(0));
+        element2.emplace_back(input.at(1));
+
+        //call merging of three vectors and then return  sorted vector
+
+        mergeVersion3(input,element1,element2,no_element);
+        return input;
+    }
+    else
+    {   int left_size{0},right_size{0},mid_size{0};
+        if(input.size()%3 == 0)
+        {
+            left_size = int(input.size()/3);
+            mid_size = int(input.size()/3);
+            right_size = int(input.size()/3);
+        }
+        else if(input.size()%3==1)
+        {
+            left_size = int(input.size()/3) + 1;
+            mid_size = int(input.size()/3);
+            right_size = int(input.size()/3);
+        }
+        else if(input.size()%3 == 2)
+        {
+            left_size = int(input.size()/3) + 1;
+            mid_size = int(input.size()/3);
+            right_size = int(input.size()/3) + 1;
+        }
+
+        // dividing original vector into three vectors left_vec,mid_vec,right_vec
+
+        vector<int> left_vec,right_vec,mid_vec;
+        for (int i = 0; i <left_size ; ++i)
+        {
+            left_vec.emplace_back(input[i]);
+        }
+        for (int i = 0; i <mid_size ; ++i)
+        {
+            mid_vec.emplace_back(input[i+left_size]);
+        }
+        for (int i = 0; i <right_size ; ++i)
+        {
+            right_vec.emplace_back(input[i+left_size+mid_size]);
+        }
+
+        //dividing the three vectors recursively
+
+        mergeSort3ways(left_vec);
+        mergeSort3ways(mid_vec);
+        mergeSort3ways(right_vec);
+
+        // sorting the vector
+
+        mergeVersion3(input,left_vec,mid_vec,right_vec);
+        return input;
+    }
+
+}
+
+// Driver Code
+
+int main()
+{
+    int n{0},temp{0};
```

Comment:
what does n{0} signify here? Can we not have int n, temp; instead ?

Reply:
n{0}, it is just a way of creating a variable n of type int and then initialise it to 0 
int n = 0, this means to create a variable of type int and then it is initialised with some garbage value first and then assign 0 to it;
so I believe n{0} is a good practice instead
So do I need to change anything there ?